db/migrations/versions/030_project_geospatial_data.py

```diff
@@ -0,0 +1,40 @@
+"""data migration populating project_geospatial_association table
+
+Revision ID: 030_project_geospatial_data
+Revises: 029_project_geospatial_m2m
+Create Date: 2024-05-28 10:20:41.204961
+
+"""
+
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "030_project_geospatial_data"
+down_revision = "029_project_geospatial_m2m"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+
+    op.execute(
+        """
+        INSERT INTO project_geospatial_association (project_id, geospatial_id)
+        SELECT DISTINCT unnested_project_dim.id, geospatial_dim.id
+        FROM (
+            SELECT id AS id, unnest(postcodes) AS postcode
+            FROM project_dim
+        ) as unnested_project_dim
+        JOIN geospatial_dim ON geospatial_dim.postcode_prefix = substring(unnested_project_dim.postcode FROM '^[A-Z]+')
+        ON CONFLICT DO NOTHING;
+        """
+    )
+
+    # ### end Alembic commands ###
+
+
+def downgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    pass
+    # ### end Alembic commands ###
```

Comment:
Migration 030 is the only addition to what's in the main https://github.com/communitiesuk/funding-service-design-post-award-data-store/pull/506